### PR TITLE
Add configurations

### PR DESCRIPTION
**Pull request by DanielMwendwa**

### CODE REVIEW

File: configuration/coding_plans.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                           coded_field="age_category_coded",
+                           analysis_file_key="age_category",
+                           fold_strategy=FoldStrategies.assert_label_ids_equal
+                       )
+                   ],
+                   code_imputation_function=code_imputation_functions.impute_age_category,
+                   ws_code=CodeSchemes.WS_CORRECT_DATASET.get_code_with_match_value("age"),
+                   raw_field_fold_strategy=FoldStrategies.assert_equal),
+
+        CodingPlan(raw_field="recently_displaced_raw",
```
Comment (IsaackMwenda) [inline configuration/coding_plans.py:157]: @as2388 please check with RDA wether they still want to drop this demog question too

File: configuration/pipeline_config.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  "RawDataSources": [
+    {
+      "SourceType": "RapidPro",
+      "Domain": "textit.in",
+      "TokenFileURL": "gs://avf-credentials/csap-text-it-token.txt",
+      "ActivationFlowNames": [
+        "csap_eu_pcve_s01e01_activation"
+      ],
+      "SurveyFlowNames": [
+        "csap_eu_pcve_demog"
```
Comment (IsaackMwenda) [inline configuration/pipeline_config.json:12]: @as2388 cc @DanielMwendwa  do we have g-cloud demog archives for previous projects that we should add here or should we use the textit versions.

**Review by IsaackMwenda — APPROVED** [review]

File: configuration/pipeline_config.json (hunk 2 above)
Comment (as2388) [reply configuration/pipeline_config.json:12]: Yes, we basically need to take the configuration from USAID-IBTCI and replace the activation flows there with the ones here (so the configuration file ends up being very long)

File: configuration/coding_plans.py (hunk 1 above)
Comment (as2388) [reply configuration/coding_plans.py:157]: have asked on the channel, will wait for a response before merging

**Review by as2388 — CHANGES_REQUESTED** [review]
Thanks a lot @DanielMwendwa, let me know if you'd like me to address any of the comments here if you'd like to

**Comment by DanielMwendwa** [comment]
> Thanks a lot @DanielMwendwa, let me know if you'd like me to address any of the comments here or if you'd like to



> Thanks a lot @DanielMwendwa, let me know if you'd like me to address any of the comments here or if you'd like to

I would like to address the comments.

File: configuration/pipeline_config.json (hunk 2 above)
Comment (DanielMwendwa) [reply configuration/pipeline_config.json:12]: @as2388 kindly work on that, thanks.

Comment (DanielMwendwa) [reply configuration/pipeline_config.json:12]: thanks, updated.

File: configuration/code_schemes.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        firebase_map = json.load(f)
+        return CodeScheme.from_firebase_map(firebase_map)
+
+
+class CodeSchemes:
+    RQA_S01E01 = _open_scheme("rqa_s01e01.json")
+
+    SOMALIA_OPERATOR = _open_scheme("somalia_operator.json")
+    ENGAGEMENT_TYPE = _open_scheme("engagement_type.json")
+    HOUSEHOLD_LANGUAGE = _open_scheme("household_language.json")
```
[the remark this answers is not in the document]
Comment (DanielMwendwa) [reply configuration/code_schemes.py]: Thanks, updated.

File: configuration/pipeline_config.json [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    "TableName": "ADSS",
+    "UuidPrefix": "avf-phone-uuid-"
+  },
+  "OperationsDashboard": {
+    "FirebaseCredentialsFileURL": "gs://avf-credentials/avf-dashboards-firebase-adminsdk-gvecb-ef772e79b6.json"
+  },
+  "SourceKeyRemappings": [
+    {"SourceKey": "avf_phone_id", "PipelineKey": "uid"},
+
+    {"SourceKey": "Rqa_EU_PCVE_S01E01 (Text) - csap_eu_pcve_s01e01_activation", "PipelineKey": "rqa_s01e01_raw", "IsActivationMessage": true},
```
Comment (as2388) [inline configuration/pipeline_config.json:66]:
```suggestion
    {"SourceKey": "Rqa_Eu_Pcve_S01E01 (Text) - csap_eu_pcve_s01e01_activation", "PipelineKey": "rqa_s01e01_raw", "IsActivationMessage": true},
```

(rapid pro mangles the case of variable names, so it should look like this instead, which you would have seen if you had tested)

File: configuration/pipeline_config.json [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    {"SourceKey": "Livelihood (Time) - csap_s08_demog", "PipelineKey": "livelihood_time"},
+    {"SourceKey": "Hh_Language (Text) - csap_s08_demog", "PipelineKey": "household_language_raw"},
+    {"SourceKey": "Hh_Language (Time) - csap_s08_demog", "PipelineKey": "household_language_time"},
+    
+    {"SourceKey": "Csap_S08_Have_Voice (Text) - csap_s08_evaluation", "PipelineKey":  "s08_have_voice_raw"},
+    {"SourceKey": "Csap_S08_Have_Voice (Time) - csap_s08_evaluation", "PipelineKey":  "s08_have_voice_time"},
+    {"SourceKey": "Csap_S08_Suggestions (Text) - csap_s08_evaluation", "PipelineKey":  "s08_suggestions_raw"},
+    {"SourceKey": "Csap_S08_Suggestions (Time) - csap_s08_evaluation", "PipelineKey":  "s08_suggestions_time"}
+  ],
+  "ProjectStartDate": "2021-08-21T00:00:00+03:00",
```
Comment (as2388) [inline configuration/pipeline_config.json:204]: Delayded to 08-29

**Review by as2388 — APPROVED** [review]
Thanks, this looks good with minor tweaks

File: configuration/pipeline_config.json [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+    {"SourceKey": "Gender (Text) - csap_eu_pcve_demog", "PipelineKey": "gender_raw"},
+    {"SourceKey": "Gender (Time) - csap_eu_pcve_demog", "PipelineKey": "gender_time"},
+    {"SourceKey": "Age (Text) - csap_eu_pcve_demog", "PipelineKey": "age_raw"},
+    {"SourceKey": "Age (Time) - csap_eu_pcve_demog", "PipelineKey": "age_time"},
+    {"SourceKey": "Location (Text) - csap_eu_pcve_demog", "PipelineKey": "location_raw"},
+    {"SourceKey": "Location (Time) - csap_eu_pcve_demog", "PipelineKey": "location_time"},
+    {"SourceKey": "Recently Displaced (Text) - csap_eu_pcve_demog", "PipelineKey": "recently_displaced_raw"},
+    {"SourceKey": "Recently Displaced (Time) - csap_eu_pcve_demog", "PipelineKey": "recently_displaced_time"},
+    {"SourceKey": "Hh_Language (Text) - csap_eu_pcve_demog", "PipelineKey": "household_language_raw"},
+    {"SourceKey": "Hh_Language (Time) - csap_eu_pcve_demog", "PipelineKey": "household_language_time"},
```
[the remark this answers is not in the document]
Comment (DanielMwendwa) [reply configuration/pipeline_config.json:78]: Noted, thanks.

**Pull request merged**